src/ape_http/providers.py

```diff
@@ -134,12 +270,70 @@ def send_transaction(self, txn: TransactionAPI) -> ReceiptAPI:
         try:
             txn_hash = self._web3.eth.send_raw_transaction(txn.encode())
         except ValueError as err:
-            raise get_tx_error_from_web3_value_error(err) from err
+            raise _get_tx_error(err) from err
+
+        receipt = self.get_transaction(txn_hash.hex())
+
+        if receipt.status == TransactionStatusEnum.FAILING and receipt.gas_used == txn.gas_limit:
+            raise OutOfGasError()
```

Comment:
For geth, this is the only OutOfGasErrors occur. Other providers raise them normally. I have never seen any other errors come this far for any other reason in geth.. Reverts / other VM errors get caught and handled above.